docs/src/get-started/concepts.rst

```diff
@@ -0,0 +1,239 @@
+  generally different irreducible representations of the symmetry group)
+  separately. This is the case of the ``spherical_harmonics_l`` key dimension
+  produced by `rascaline`_: different blocks will contain the :math:`\ell = 1`
+  and :math:`\ell = 2` parts of an equivariant representation.
+
+.. _rascaline: https://github.com/Luthaf/rascaline/
+
+.. _concept-Labels:
+
+Labels
+^^^^^^
+
+A fundamental part of equistore is to carry simultaneously the data used in
+machine learning and the associated metadata. The first kind of metadata we
+encountered was the keys of a :py:class:`TensorMap`, stored as an instance of
+the :py:class:`Labels` class. This class is also used to store all other
+metadata in equistore, i.e. all the metadata associated with a given
+:py:class:`TensorBlock`.
+
+
+.. _fig-labels:
+
+.. figure:: ../../static/images/Labels.*
+    :width: 600px
+    :align: center
+
+    Illustration of two different :py:class:`Labels` instances, corresponding to
+    potential *samples* (green, on the left) and *properties* (red, on the
+    right) of a :py:class:`TensorBlock`.
+
+
+A set of :py:class:`Labels` can be seen as a two dimensional array of integers,
+where each row corresponds to an entry in the data, and each column is a
+*dimension*, which is named. For example, in the illustration above, the set of
+Labels on the left has two dimensions (``structure`` and ``center``), and 10
+entries (10 rows); while the Labels on the right has four dimensions and 8
+entries. Depending on the language you use, :py:class:`Labels` entries and
+dimensions' names can be accessed and manipulated in different ways, please
+refer to the corresponding :ref:`API documentation <api-reference>` for more
+information.
+
+.. _concept-TensorBlock:
+
+TensorBlock
+^^^^^^^^^^^
+
+The final core object of equistore is the :py:class:`TensorBlock`, containing a
+dense array of data and metadata describing the different axes of this array.
+The simplest possible TensorBlock is represented below, and contains three things:
+
+- a 2-dimensional data array;
+- metadata describing the rows of this array, called **samples** and stored as a
+  set of :py:class:`Labels`;
+- metadata describing the columns of this array, called **properties**, also
+  stored as a set of :py:class:`Labels`.
+
+The samples store information about **what objects** does the data represent,
+while properties store information about **how** these objects are represented.
+Taking a couple of examples for clarity:
+
+- if a TensorBlock is storing an atom-centered machine learning representation,
+  the samples would contain the index of the atom ``"center"`` and the index of
+  the corresponding ``"structure"``; while the properties would contain
+  information about the e.g. the basis functions used to define the
+  representation. The :ref:`Labels figure <fig-labels>` above contains an
+  example of samples and properties that one would find in machine learning
+  representation.
```

Comment:
Can you start with a simpler example that this? It's a bit "being thrown into the deep end" on a complex thought